app/assets/javascripts/discourse/app/components/user-field.js

```diff
@@ -1,11 +1,33 @@
 import discourseComputed from "discourse-common/utils/decorators";
 import Component from "@ember/component";
 import { fmt } from "discourse/lib/computed";
+import { action } from "@ember/object";
 
 export default Component.extend({
   classNameBindings: [":user-field", "field.field_type", "customFieldClass"],
   layoutName: fmt("field.field_type", "components/user-fields/%@"),
 
+  didInsertElement() {
+    this.field.component = this;
```

Comment:
I don't know much all this code, but it looks like a smell to me to carry this component like this.

Also I see it's used as "element" later, when most of the times these elements are DOM nodes, and not Ember components.